lib/fs.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const metasync = require('metasync');
+
+const error = (err, options, cb) => {
+  if (err.code === 'EEXIST') cb();
+  else if (!options.errorLog) cb(err);
+  else {
+    options.errorLog.write(JSON.stringify(err));
+    cb();
+  }
+};
+
+const cpDir = (src, dest, options, cb, curDepth) => {
+  if (curDepth > options.depth) return;
+  fs.mkdir(dest, err => {
+    if (err !== null && err.code !== 'EEXIST') {
+      error(err, options, cb);
+      return;
+    }
+    fs.readdir(src, (err, files) => {
+      if (err) {
+        error(err, options, cb);
+        return;
+      }
+      metasync.each(
+        files,
+        (file, cb) => {
+          if (
+            (options.filter instanceof RegExp && !options.filter.test(file)) ||
+            (typeof options.filter === 'function' && !options.filter(file)) ||
+            typeof options.filter === 'undefined'
+          ) {
+            fs.stat(path.join(src, file), (err, stats) => {
+              if (err) {
+                error(err, options, cb);
+                return;
+              }
+              const srcFile = path.join(src, file);
+              const destFile = path.join(dest, file);
+              if (stats.isDirectory())
+                cpDir(srcFile, destFile, options, cb, ++curDepth);
```

Comment:
```suggestion
                cpDir(srcFile, destFile, options, cb, curDepth + 1);
```
If you have more than one nested directory on the same level it will increment `curDepth` for each directory, which will lead to incorrect behaviour.